src/transformers/models/siglip/modeling_siglip.py

```diff
@@ -543,6 +825,33 @@ def _init_weights(self, module):
             module.bias.data.zero_()
             module.weight.data.fill_(1.0)
 
+    @classmethod
+    def _autoset_attn_implementation(
```

Comment:
c.f. https://github.com/huggingface/transformers/pull/31203#issuecomment-2145036503

I let this slip in for IDEFICS2, but it never should have been included